gma/src/android/gma_android.cc

```diff
@@ -298,7 +294,11 @@ Future<AdapterInitializationStatus> Initialize(JNIEnv* env, jobject activity,
     return Future<AdapterInitializationStatus>();
   }
 
-  if (!util::Initialize(env, activity)) {
+  // GMA requires Google Play services if the class
+  // "com.google.android.gms.ads.internal.ClientApi" does not exist.
+  if (!util::FindClass(env, "com/google/android/gms/ads/internal/ClientApi") &&
+      google_play_services::CheckAvailability(env, activity) !=
+          google_play_services::kAvailabilityAvailable) {
     if (init_result_out) {
```

Comment:
In this scenario, since util::Initialize was previously called, we need to call util::Terminate to ensure that the refcount is correct.

Reply:
Good point. I see that Util::Terminate is called in another fail case later in this function, so I will call it in this fail case as well. Done.